addons/onboarding/models/onboarding_progress.py

```diff
@@ -23,9 +23,20 @@ class OnboardingProgress(models.Model):
     onboarding_id = fields.Many2one(
         'onboarding.onboarding', 'Related onboarding tracked', required=True, ondelete='cascade')
     progress_step_ids = fields.One2many('onboarding.progress.step', 'progress_id', 'Progress Steps Trackers')
+    not_per_company = fields.Selection(
+        [('True', 'True')], string='Only one', compute='_compute_not_per_company', store=True,
+        help="Should there only be one record for this onboarding on this database."
+    )
+
     _sql_constraints = [
-        ('onboarding_company_uniq', 'unique (onboarding_id,company_id)',
+        ('onboarding_company_uniq', 'UNIQUE (onboarding_id, company_id)',
          'There cannot be multiple records of the same onboarding completion for the same company.'),
+        ('onboarding_not_per_company_unique', 'UNIQUE (onboarding_id, not_per_company)',
+         'There cannot be multiple completion records for one-per-database onboardings.'),
+        ('not_per_company_true_wo_company',
+         "CHECK ((company_id IS NULL AND not_per_company = 'True')"
+         'OR (company_id IS NOT NULL AND not_per_company IS NULL))',
+         'not_per_company must be "True" for one-per-company onboardings and unset otherwise.'),
```

Comment:
I think it would be worth to add a comment mentioning the UNIQUE + null behavior